clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -761,6 +761,10 @@ bool ConstStructBuilder::Build(const InitListExpr *ILE, bool AllowOverwrite) {
       if (Field->hasAttr<NoUniqueAddressAttr>())
         AllowOverwrite = true;
     } else {
+      llvm::Type *LoadType = CGM.getTypes().convertTypeForLoadStore(
```

Comment:
I have no idea what this is supposed to do; why don't we want to use the "memory" type here?

Reply:
In `ConstantEmitter::emitForMemory` we now split constant into individual bytes if the target type is a long `_BitInt`. However after an integer constant expected by `AppendBitField`, so this just folds individual bytes back to an integer.

Reply:
Oh, I see... please move the duplicated code into the implementation of AppendBitField, and add a comment explaining it.

Reply:
That would change `AppendBitField` to have a `llvm::Value` parameter instead of `llvm::ConstantInt`, is that still ok?

Reply:
I think you can use llvm::Constant? But that's fine, sure.